core/src/Revolution/Error/modErrorHandler.php

```diff
@@ -58,7 +58,7 @@ function __construct(modX &$modx, array $stack = [])
      */
     public function handleError($errno, $errstr, $errfile = null, $errline = null, $errcontext = null)
     {
-        if (error_reporting() == 0) {
+        if (error_reporting() == 0 || ($errno === E_WARNING && (error_reporting() & E_WARNING) === 0 )) {
```

Comment:
I believe this needs to be `!(error_reporting() & $errno)` so that suppression works with all non-fatal errors, not only E_WARNING. For example, E_NOTICE would not be suppressed using your solution.